packages/components/src/components/drawer/drawer.jsx

```diff
@@ -15,87 +15,73 @@ const IconDrawer = ({ className }) => (
+            }}
+        >
             <div
-                className={classNames('dc-drawer', className, {
-                    [`dc-drawer--${anchor}`]: !is_mobile,
-                    'dc-drawer--open': this.state.is_open,
+                className={classNames('dc-drawer__toggle', {
+                    'dc-drawer__toggle--open': is_open,
                 })}
-                style={{
-                    zIndex,
-                    transform:
-                        this.state.is_open &&
-                        !is_mobile &&
-                        (anchor === 'left'
-                            ? `translateX(calc(${width}px - 16px))`
-                            : `translateX(calc(-${width}px + 16px))`),
-                }}
+                onClick={toggleDrawer}
             >
-                <div
-                    className={classNames('dc-drawer__toggle', {
-                        'dc-drawer__toggle--open': this.state.is_open,
-                    })}
-                    onClick={this.toggleDrawer}
-                >
-                    {is_mobile ? (
-                        <Icon icon='IcChevronUp' className='dc-drawer__toggle-icon' />
-                    ) : (
-                        <IconDrawer
-                            className={classNames('dc-drawer__toggle-icon', {
-                                [`dc-drawer__toggle-icon--${anchor}`]: !is_mobile,
-                            })}
-                        />
-                    )}
-                </div>
-                <div
-                    className={classNames('dc-drawer__container', { [`dc-drawer__container--${anchor}`]: !is_mobile })}
-                >
-                    {header && <div className='dc-drawer__header'>{header}</div>}
-                    <div className={classNames('dc-drawer__content', contentClassName)}>{children}</div>
-                    {footer && <div className='dc-drawer__footer'>{footer}</div>}
-                </div>
+                {is_mobile ? (
+                    <Icon icon='IcChevronUp' className='dc-drawer__toggle-icon' />
+                ) : (
+                    <IconDrawer
+                        className={classNames('dc-drawer__toggle-icon', {
+                            [`dc-drawer__toggle-icon--${anchor}`]: !is_mobile,
+                        })}
+                    />
+                )}
             </div>
-        );
-    }
-}
+            <div className={classNames('dc-drawer__container', { [`dc-drawer__container--${anchor}`]: !is_mobile })}>
+                {header && <div className='dc-drawer__header'>{header}</div>}
+                <div className={classNames('dc-drawer__content', contentClassName)}>{children}</div>
+                {footer && <div className='dc-drawer__footer'>{footer}</div>}
+            </div>
+        </div>
+    );
+};
 
 Drawer.propTypes = {
     children: PropTypes.oneOfType([PropTypes.array, PropTypes.object]),
```

Comment:
could you please check propTypes we don't have some of them any more

Reply:
yes, updated the propTypes